token-standard/cli/src/apis/ledger-api-utils.ts

```diff
@@ -322,3 +335,70 @@ function signTransaction(
     signedHash,
   };
 }
+
+interface Completion {
+  updateId: string;
+  // the openAPI definition claims these two can be null
+  synchronizerId?: string;
+  recordTime?: string;
+}
+
+/**
+ * Polls the completions endpoint until
+ * the completion with the given (userId, commandId, submissionId) is returned.
+ * Then returns the updateId, synchronizerId and recordTime of that completion.
+ */
+async function awaitCompletion(
```

Comment:
should there be a timeout or something on this after 2min maybe? (with 1min being the local participant timeout where you would definitely expect _something_ but your connection with the partipant might die so timing out on the client side still seems reasonable)